portal-api/src/v1/controllers/deal-submission-details.controller.js

```diff
@@ -149,21 +155,21 @@ exports.update = (req, res) => {
     submissionDetails = await checkAllCountryCodes(deal, submissionDetails);
 
     if (submissionDetails.supplyContractCurrency) {
-      submissionDetails.supplyContractCurrency = await checkCurrency(
-        deal.supplyContractCurrency,
-        submissionDetails.supplyContractCurrency,
-      );
+      submissionDetails.supplyContractCurrency = await checkCurrency(deal.supplyContractCurrency, submissionDetails.supplyContractCurrency);
     }
 
     const dealAfterAllUpdates = await updateSubmissionDetails(req.params.id, submissionDetails, user);
 
-    const validationErrors = validateSubmissionDetails({ ...dealAfterAllUpdates.submissionDetails, ...req.body });
+    const validationErrors = await validateSubmissionDetails({ ...dealAfterAllUpdates.submissionDetails, ...req.body });
 
     const response = {
       validationErrors,
       data: dealAfterAllUpdates.submissionDetails,
     };
 
     return res.status(200).json(response);
-  });
+  } catch (error) {
+    console.error(error);
+    return res.status(500).send();
```

Comment:
Don't quote me on this -- but do unhandled errors default to `500`s? https://expressjs.com/en/guide/error-handling.html

I'm happy to have this explicit if you'd like :)

Reply:
I don't think so, however we have been using `500` for catch-all errors in the project.